src/API/Google/BudgetRecommendations.php

```diff
@@ -69,29 +69,34 @@ public function get_recommendations( array $country_codes ): ?array {
 		}
 
 		$location_ids = $this->get_location_ids( $country_codes );
-
-		$request = new GenerateRecommendationsRequest(
-			[
-				'customer_id'                => $this->options->get_ads_id(),
-				'merchant_center_account_id' => $this->options->get_merchant_id(),
-				'recommendation_types'       => [ RecommendationType::CAMPAIGN_BUDGET ],
-				'advertising_channel_type'   => AdvertisingChannelType::PERFORMANCE_MAX,
-				'positive_locations_ids'     => array_keys( $location_ids ),
-				'country_codes'              => $country_codes,
-				'bidding_info'               => new BiddingInfo(
+		$merchant_id  = $this->options->get_merchant_id();
+
+		$request_data = [
+			'customer_id'              => $this->options->get_ads_id(),
+			'recommendation_types'     => [ RecommendationType::CAMPAIGN_BUDGET ],
+			'advertising_channel_type' => AdvertisingChannelType::PERFORMANCE_MAX,
+			'positive_locations_ids'   => array_keys( $location_ids ),
+			'country_codes'            => $country_codes,
+			'bidding_info'             => new BiddingInfo(
+				[
+					'bidding_strategy_type' => BiddingStrategyType::MAXIMIZE_CONVERSION_VALUE,
+				]
+			),
+			'asset_group_info'         => [
+				new AssetGroupInfo(
 					[
-						'bidding_strategy_type' => BiddingStrategyType::MAXIMIZE_CONVERSION_VALUE,
-					]
+						'final_url' => $this->get_site_url(),
+					],
 				),
-				'asset_group_info'           => [
-					new AssetGroupInfo(
-						[
-							'final_url' => $this->get_site_url(),
-						],
-					),
-				],
-			]
-		);
+			],
+		];
+
+		// Only include merchant_center_account_id if Merchant Center account is connected.
+		if ( $merchant_id > 0 ) {
+			$request_data['merchant_center_account_id'] = $merchant_id;
+		}
```

Comment:
Shall we use array union expression instead of `if` condition?

```php
  $request_data += ( $merchant_id > 0 ) ? [ 'merchant_center_account_id' => $merchant_id ] : [];
```